src/odemis/acq/millmng.py

```diff
@@ -0,0 +1,526 @@
+                                                           iteration=self._iterations[setting_nb],
+                                                           duration=self._pass_duration[setting_nb],
+                                                           probe_size=probe_size, overlap=overlap)
+                try:
+                    self._future.running_subf.result()
+                    logging.debug(f"Milling finished for feature: {site.name.value}")
+                except Exception as exp:
+                    logging.debug(
+                        f"Milling setting: {milling_settings.name.value} for feature: {site.name.value} failed: {exp}")
+                    raise
+
+            else:
+                for itr in range(self._iterations[setting_nb]):
+                    with self._future._task_lock:
+                        if self._future._task_state == CANCELLED:
+                            raise CancelledError()
+
+                    # restore the milling current
+                    self._scanner.probeCurrent.value = milling_settings.current.value
+
+                    self._future.running_subf = mill_rectangle(milling_settings.roi.value, self._scanner,
+                                                               iteration=self._iterations[setting_nb],
+                                                               duration=self._pass_duration[setting_nb],
+                                                               probe_size=probe_size, overlap=overlap)
+                    try:
+                        self._future.running_subf.result()
+                        logging.debug(f"Milling finished for feature: {site.name.value}")
+                    except Exception as exp:
+                        logging.debug(
+                            f"Milling setting: {milling_settings.name.value} for feature: {site.name.value} at iteration: {itr} failed: {exp}")
+                        raise
+
+                    # Change the current to the drift correction current
+                    self._scanner.probeCurrent.value = milling_settings.dcCurrent.value
+                    # Acquire an image at the given location (RoI)
+                    drift_est.acquire()
+
+                    # Estimate the drift
+                    drift_est.estimate()
+                    drift = (pxs_anchor[0] * drift_est.drift[0],
+                             -(pxs_anchor[1] * drift_est.drift[1]))
+
+                    # Move FIB to compensate drift
+                    previous_shift = self._scanner.shift.value
+                    logging.debug("Ion-beam shift in m : %s", previous_shift)
+                    self._scanner.shift.value = (drift[0] + previous_shift[0],
+                                                 drift[1] + previous_shift[1])  # shift in m - absolute position
+                    logging.debug("New Ion-beam shift in m : %s", self._scanner.shift.value)
```

Comment:
Can you merge the logging lfor the previous and new ion-beam shifts into a single logging line? That'd make it more compact in the logs.